### DISCUSSION TITLE
namespace for VtkRemoteLocalView

### DISCUSSION THREAD

**Discussion by Jimmy-KL** [Kitware/trame#164] [Q&A] [answered]
In the example [DynamicLocalRemoteRendering.py](https://github.com/Kitware/trame/blob/master/examples/06_vtk/02_ContourGeometry/DynamicLocalRemoteRendering.py),  I have 2 questions about the following code.
`view = vtk_widgets.VtkRemoteLocalView(`
               ` renderWindow,`
                `namespace="demo",`
               ` mode=("override === 'auto' ? demoMode : override", "local"),`
           ` )`
I have checked the [API](https://trame.readthedocs.io/en/latest/trame.widgets.vtk.html#trame.widgets.vtk.VtkRemoteLocalView), but it's not clear to me.  What is the meaning of `namespace` and the value of `demoMode` here? demoMode is not defined in the script.

**Comment by jourdain** [comment, marked as answer]
[Namespace is the prefix for all the variables/triggers](https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/widgets/vtk/common.py#L340):
```python
namespace=...,           # Prefix for variables. See below. (required)
```
[Here is a list of state variables](https://github.com/Kitware/trame-vtk/blob/master/trame_vtk/widgets/vtk/common.py#L359-L361) that get generated based on that namespace.

- {namespace}Mode
- {namespace}Scene
- {namespace}Id
